src/main/java/Duke.java

```diff
@@ -31,49 +32,49 @@ public static void main(String[] args) {
-                    splitCommand = command.split(" ", 2);
-                    date = splitCommand[1].split("/at", 2);
-                    Event event = new Event(date[0].trim(), date[1].trim());
-                    tasks[numberOfTasks] = event;
-                    numberOfTasks++;
-                    System.out.println(" Alright, I've added this task:\n   " + event.toString() + "\n"
-                            + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
-                    break;
-                case "todo":
-                    toDo = new ToDo(command.substring(5, command.length()));
-                    tasks[numberOfTasks] = toDo;
-                    numberOfTasks++;
-                    System.out.println(" Alright, I've added this task:\n   " + toDo.toString() + "\n"
-                            + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
-                    break;
-                default:
-                    toDo = new ToDo(command);
-                    tasks[numberOfTasks] = toDo;
-                    numberOfTasks++;
-                    System.out.println(" Alright, I've added this task:\n   " + toDo.toString() + "\n"
-                            + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
-                    break;
-                }
+                continue;
+            }
+            String[] tokens = command.split(" ");
+            switch (tokens[0]) {
+            case "done":
+                int taskNumber = Integer.parseInt(tokens[1]);
+                task= tasks[taskNumber-1];
+                task.markAsDone();
+                System.out.println(HORIZONTAL_LINE + "\n Nice! I've marked this task as done:");
+                System.out.println("   " + task.toString()  + "\n" + HORIZONTAL_LINE);
+                break;
+            case "deadline":
+                splitCommand = command.split(" ", 2);
+                date = splitCommand[1].split("/by", 2);
+                Deadline deadline = new Deadline(date[0].trim(), date[1].trim());
+                tasks[numberOfTasks] = deadline;
+                numberOfTasks++;
+                System.out.println(" Alright, I've added this task:\n   " + deadline.toString() + "\n"
+                        + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
+                break;
+            case "event":
+                splitCommand = command.split(" ", 2);
+                date = splitCommand[1].split("/at", 2);
+                Event event = new Event(date[0].trim(), date[1].trim());
+                tasks[numberOfTasks] = event;
+                numberOfTasks++;
+                System.out.println(" Alright, I've added this task:\n   " + event.toString() + "\n"
+                        + " Now you have " + numberOfTasks + " tasks in your list.\n" + HORIZONTAL_LINE);
+                break;
+            case "todo":
```

Comment:
since the "case "todo": " and "default" have the same contents, maybe you can keep just one of them? this would make it shorter and more readable in my opinion:) but if you have different meanings for the two of them that's fine, but generally I cannot find any code violations:))

Reply:
Thank you for noting this, I will make sure to rectify that soon!